test/tasks/install_deps_test.rb

```
@@ -0,0 +1,225 @@
+      # Verify install was called for each package
+      Kompo::InstallDeps::ForMacOS::PACKAGES.each_value do |package|
+        assert @mock.called?(:run, "/opt/homebrew/bin/brew", "install", package.name),
+          "Expected brew install #{package.name} to be called"
+      end
+    ensure
+      # Clean up marker files after test
+      Kompo::InstallDeps::ForMacOS::PACKAGES.each_value do |package|
+        File.delete(package.marker_file) if File.exist?(package.marker_file)
+      end
+    end
+  end
```

RuboCop (1.82.1)

[convention] Assignment Branch Condition size for `test_installs_packages_when_not_installed` is too high. [<4, 28, 7> 29.14/23]

(Metrics/AbcSize)